app/models/conversation.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: conversations
+#
+#  id         :bigint           not null, primary key
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Conversation < ApplicationRecord
+  has_many :conversation_users, dependent: :destroy
+  has_many :users, through: :conversation_users
+  has_many :messages, dependent: :destroy
+
+  def self.create_conversations(matching_users)
+    user = matching_users.pop
+    matching_users.each do |match|
+      conversation = share_conversation(user, match)
+      if conversation.blank?
+        conversation = Conversation.create!
+        user.conversations << conversation
+        match.conversations << conversation
+      end
+      conversation.messages.create!(
+        text: I18n.t('api.messages.target_match')
+      )
+    end
+  end
+
+  def self.share_conversation(user, match)
+    return if user.conversations.empty? || match.conversations.empty?
+
+    user.conversations.each do |conv|
+      return conv if match.conversations.include?(Conversation.find(conv.id))
+    end
+
+    nil
+  end
+
+  def not_current_user(current_user)
+    users.reject { |user| user == current_user }
+  end
```

Comment:
You can use a scope here too.
For example: In the User class: `:not_id, -> (user_id) { where.not(id: user_id) }`. Then you can call `users.not_id(current_user)`. With this approach, you will query the database directly.

Reply:
Here the `not_current_user`  refers to the rest of the users on the conversation that are not current user. For now is just one but it opens up the possibility to be more in the future.
This is used mainly on conversation views, the idea is to show within the conversation, the list of users that are part of it, not showing current user. 
Maybe I can still do a scope, but it would be definitely tricky.